pkg/cloudprovider/providers/azure/azure_storage.go

```diff
@@ -19,53 +19,80 @@ package azure
+	if len(accountName) == 0 {
+		// find a storage account that matches accountType
+		accounts, err := az.getStorageAccounts(accountType)
+		if err != nil {
+			return "", "", fmt.Errorf("could not list storage accounts for account type %s: %v", accountType, err)
 		}
-	}
-	for _, account := range accounts {
-		glog.V(4).Infof("account %s type %s location %s", account.Name, account.StorageType, account.Location)
-		if ((storageType == "" || account.StorageType == storageType) && (location == "" || account.Location == location)) || len(storageAccount) > 0 {
-			// find the access key with this account
-			key, innerErr := az.getStorageAccesskey(account.Name)
-			if innerErr != nil {
-				errResult = fmt.Errorf("could not get storage key for storage account %s: %v", account.Name, innerErr)
-				continue
+
+		if len(accounts) > 0 {
+			for _, account := range accounts {
+				glog.V(4).Infof("account %s type %s location %s", account.Name, account.StorageType, account.Location)
+				if (accountType == "" || account.StorageType == accountType) && (location == "" || account.Location == location) {
+					glog.V(4).Infof("found a matching account %s type %s location %s", account.Name, account.StorageType, account.Location)
+					accountName = account.Name
+					break
+				}
 			}
+		}
 
-			if innerErr = az.createFileShare(account.Name, key, name, requestGB); innerErr != nil {
-				errResult = fmt.Errorf("failed to create share %s in account %s: %v", name, account.Name, innerErr)
-				continue
+		if len(accountName) == 0 {
+			// not found a matching account, now create a new account in current resource group
+			accountName = generateStorageAccountName(fileShareAccountNamePrefix)
+			if location == "" {
+				location = az.Location
+			}
+			if accountType == "" {
+				accountType = defaultStorageAccountType
+			}
+
+			glog.V(2).Infof("azureFile - no storage account found, begin to create a new storage account %s in resource group %s, location: %s, accountType: %s",
+				accountName, az.ResourceGroup, location, accountType)
+			cp := storage.AccountCreateParameters{
+				Sku:      &storage.Sku{Name: storage.SkuName(accountType)},
+				Tags:     &map[string]*string{"created-by": to.StringPtr("azure-file")},
+				Location: &location}
+			cancel := make(chan struct{})
+
+			_, errchan := az.StorageAccountClient.Create(az.ResourceGroup, accountName, cp, cancel)
+			err := <-errchan
+			if err != nil {
+				return "", "", fmt.Errorf(fmt.Sprintf("Failed to create storage account %s, error: %s", accountName, err))
 			}
-			glog.V(4).Infof("created share %s in account %s", name, account.Name)
-			return account.Name, key, nil
 		}
 	}
 
-	if errResult == nil {
-		errResult = fmt.Errorf("failed to find a matching storage account")
+	// find the access key with this account
+	accountKey, err := az.getStorageAccesskey(accountName)
+	if err != nil {
```

Comment:
Combine err check in same line

Reply:
if I code like following, it looks ugly, since accountKey will be used in the next statement.
```
if accountKey, err := az.getStorageAccesskey(accountName); err != nil {
...
} else {
   if err := az.createFileShare(accountName, accountKey, shareName, requestGiB); err != nil {
}
```